wf-galleon/WFLY-19021-Stability_In_Provisioning.adoc

```diff
@@ -0,0 +1,294 @@
+
+Expected config stability levels for WildFly feature-packs:
+
+* wildfly-ee-galleon-pack: `community`
+* wildfly-galleon-pack: `community`
+* wildfly-preview-feature-pack: `preview`
+* wildfly-core: `community`
+
+Expected package stability levels for WildFly feature-packs:
+
+* wildfly-ee-galleon-pack: `experimental`
+* wildfly-galleon-pack: `experimental`
+* wildfly-preview-feature-pack: `experimental`
+* wildfly-core: `experimental`
+
+=== Requirements at provisioning time
+
+Provisioning time follows some rules to properly handle stability and avoid features/packages of wrong stability being provisioned.
+
+* The feature-pack contained config stability level is used to constrain the set of provisioned features. Any features at a lower stability level contained in the 
+feature-pack (if any), are not provisioned.
+* The feature-pack contained package stability level is used to constrain the set of provisioned packages. Any packages at a lower stability level contained in the 
+feature-pack (if any), are not provisioned.
+* By default, the config and package stability levels of each feature-pack are used when provisioning its own content. 
+* Each feature-pack present in the provisioning configuration can have a different config stability level. 
+The config stability of each feature-pack applies to its own content and doesn't leak into other feature-packs.
+* Each feature-pack present in the provisioning configuration can have a different package stability level. 
+The package stability of each feature-pack applies to its own content and doesn't leak into other feature-packs.
+* A user can specify a config stability level, by means of the `config-stability-level` Galleon option. NB: This option can't be set when `stability-level` is set.
+This stability level is used to constrain the stability level of all features present in configurations generated during provisioning. If the specified level is lower than a feature-pack config stability, 
+the feature-pack config stability level is used.
+* A user can specify a package stability level, by means of the `package-stability-level` Galleon option. 
+This stability level is used to constrain the stability level of all provisioned packages. NB: This option can't be set when `stability-level` is set. 
+If the specified level is lower than a feature-pack package stability, 
+the feature-pack package stability level is used.
+* A user can specify a stability level for both configs and packages, by means of the `stability-level` Galleon option. This option is a convenience 
+allowing to set both the config and package stability level using a single option. It can't be set when `config-stability-level` or `package-stability-level` 
+are in use.
+* At provisioning time, usage of `<stability-level>` and `<config|package-stability-level>` options in the same provisioning execution 
+is not supported and would lead to a failure.
+
+For example, a provisioning configuration containing 3 feature-packs:
```

Comment:
I think this example needs to cover something like the 'provision a preview config from a wildfly-galleon-pack' case. The example right now has no FPs with different values for the 3 stability settings, so the scenario of wanting to provision a lower config stability than the FP's setting is not described. I think it should be supported and that's why I filed https://issues.redhat.com/browse/GAL-361, but reading the spec I see it's not explicitly stated it's supported.  It's implied by statements like 'This level can be overridden at provisioning time with the Galleon option `<config-stability-level>`' but 'overridden' doesn't have to mean it can overridden to any value.

Reply:
@bstansberry , I added a clear example and specified that the default are not minimal that constrain provisioning.